onnx/utils.py

```diff
@@ -21,3 +24,138 @@ def polish_model(model):  # type: (ModelProto) -> ModelProto
     model = onnx.optimizer.optimize(model)
     onnx.checker.check_model(model)
     return model
+
+
+class Extractor:
+    def __init__(self, model):  # type: ignore
+        self.model = onnx.shape_inference.infer_shapes(model)
+        self.graph = self.model.graph
+        self.wmap = self._buildNameDict(self.graph.initializer)
+        self.vimap = self._buildNameDict(self.graph.value_info)
+
+    @staticmethod
+    def _buildNameDict(objs):  # type: ignore
+        return {obj.name: obj for obj in objs}
+
+    def _filterTensors(self, originals, new_names):  # type: ignore
+        tmap = self._buildNameDict(originals)
+        original_names = set(tmap.keys())
+        s_new_names = set(new_names)
+        names_keep = s_new_names & original_names
+        names_add = s_new_names - original_names
+
+        tensors = []
+        for name in names_keep:
+            tensors.append(tmap[name])
+        for name in names_add:
+            # activation become input or output
+            tensors.append(self.vimap[name])
+
+        # sort the tensors
+        new_tmap = self._buildNameDict(tensors)
+        sorted_tensors = [new_tmap[name] for name in new_names]
```

Comment:
why is sorting required here?

Reply:
I think it is to ensure that the generated model has inputs/outputs in the order specified by the user (via the parameters).

Reply:
Yes, this is to make the order of inputs/outputs of the generated model to be the same as user input. This is not necessary but having this won't make user supervised.